packages/big-design/src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,230 @@
+import { AddCircleOutlineIcon, RemoveCircleOutlineIcon } from '@bigcommerce/big-design-icons';
+import React, { forwardRef, useEffect, useMemo, useState, Ref } from 'react';
+
+import { useUniqueId } from '../../hooks';
+import { typedMemo, warning } from '../../utils';
+import { FormControlDescription, FormControlError, FormControlLabel } from '../Form';
+
+import { StyledCounterButton, StyledCounterInput, StyledCounterWrapper } from './styled';
+
+export interface CounterProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  label?: React.ReactChild;
+  labelId?: string;
+  description?: React.ReactChild;
+  error?: React.ReactNode | React.ReactNode[];
+  value: number;
+  step?: number;
+  onCountChange(count: number): void;
+}
+
+interface PrivateProps {
+  forwardedRef: Ref<HTMLInputElement>;
+}
+
+export const StylableCounter: React.FC<CounterProps & PrivateProps> = typedMemo(
+  ({
+    min = 0,
+    max = 100,
+    step = 1,
+    forwardedRef,
+    label,
+    labelId,
+    description,
+    error,
+    disabled,
+    value,
+    onCountChange,
```

Comment:
🤔What are your thoughts on using this vs `onChange`?

Reply:
Using something other than the native `onChange` (in addition to following the style of the `<Select />` component, for example) allows for an `onChange` to be applied directly to the `<input />` component inside the `<Counter />` parent while supplying the change logic to the parent separately. `onCountChange` also has different behavior than a native `onChange` function - it returns a number and not an event.
<img width="1408" alt="Screen Shot 2020-04-20 at 1 59 23 PM" src="https://user-images.githubusercontent.com/37593557/79789009-28c57a00-830f-11ea-8a6a-57297136068f.png">